crates/outline_panel/src/outline_panel.rs

```diff
@@ -3267,6 +3373,57 @@ impl OutlinePanel {
                                         Some(UPDATE_DEBOUNCE)
                                     };
                                     excerpt.outlines = ExcerptOutlines::Outlines(fetched_outlines);
+
+                                    // Collect outlines to check for default expansion
+                                    let outlines_to_check =
+                                        if let Some(default_depth) = pending_default_depth {
+                                            if let ExcerptOutlines::Outlines(outlines) =
+                                                &excerpt.outlines
+                                            {
+                                                if default_depth == 0 {
+                                                    // For depth 0, collapse all outlines with children
+                                                    outlines.clone()
+                                                } else {
+                                                    // For other depths, collapse outlines at or below that depth
+                                                    outlines
+                                                        .iter()
+                                                        .filter(|outline| {
+                                                            outline.depth >= default_depth
+                                                        })
+                                                        .cloned()
+                                                        .collect::<Vec<_>>()
+                                                }
+                                            } else {
+                                                Vec::new()
+                                            }
+                                        } else {
+                                            Vec::new()
+                                        };
+
+                                    Some((debounce, outlines_to_check))
```

Comment:
`debounce` can be removed from here and always used directly.

We can also do `let mut outlines_to_check = Vec::new()` and do `extend` if the settings change is correct: that will remove a lot of if/else nesting.